website/docs/enterprise/installation/airgap.mdx

```diff
@@ -0,0 +1,560 @@
+...
+```
+The different helm charts required
+
+```bash
+➜ kubectl get helmcharts.source.toolkit.fluxcd.io
+NAME                                  CHART           VERSION   SOURCE KIND      SOURCE NAME                      AGE   READY   STATUS
+flux-system-cert-manager              cert-manager    0.0.7     HelmRepository   weaveworks-charts                13h   True    pulled 'cert-manager' chart with version '0.0.7'
+flux-system-tf-controller             tf-controller   0.9.2     HelmRepository   tf-controller                    13h   True    pulled 'tf-controller' chart with version '0.9.2'
+flux-system-weave-gitops-enterprise   mccp            v0.10.2   HelmRepository   weave-gitops-enterprise-charts   13h   True    pulled 'mccp' chart with version '0.10.2'
+```
+
+And the container images
+
+```bash
+
+kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec['containers','initContainers'][*].image}" |tr -s '[[:space:]]' '\n' \
+| sort | uniq | grep -vE 'kindest|etcd|coredns'
+
+docker.io/prom/prometheus:v2.34.0
+docker.io/weaveworks/cluster-controller:v1.4.1
+docker.io/weaveworks/weave-gitops-enterprise-clusters-service:v0.10.2
+docker.io/weaveworks/weave-gitops-enterprise-ui-server:v0.10.2
+ghcr.io/fluxcd/flagger-loadtester:0.22.0
+ghcr.io/fluxcd/flagger:1.21.0
+ghcr.io/fluxcd/helm-controller:v0.23.1
+ghcr.io/fluxcd/kustomize-controller:v0.27.1
+ghcr.io/fluxcd/notification-controller:v0.25.2
+...
+```
+
+## Collect and publish artifacts
+
+This section guides you to push installed artifacts to your private registry.
+The following Makefile could help you with each of the stages.
+
+```makefile
+.PHONY := all
+
+#set these variable with your custom configuration
+PRIVATE_HELM_REPO_NAME=private
+REGISTRY=localhost:5001
+CERT_MANAGER_VERSION=0.0.8
+WGE_VERSION=0.10.2
+
+CERT_MANAGER=cert-manager-$(CERT_MANAGER_VERSION)
+CERT_MANAGER_CHART=$(CERT_MANAGER).tgz
+
+WGE=mccp-$(WGE_VERSION)
+WGE_CHART=$(WGE).tgz
+
+all: images charts
+
+charts: pull-charts push-charts
+
+images:
+	kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec['containers','initContainers'][*].image}" \
+    |tr -s '[[:space:]]' '\n' | sort | uniq | grep -vE 'kindest|kube-(.*)|etcd|coredns' | xargs -L 1 -I {} ./image-pull-push.sh {} $(REGISTRY)
+	kubectl get microvmmachinetemplates --all-namespaces -o jsonpath="{.items[*].spec.template.spec.kernel.image}"|tr -s '[[:space:]]' '\n' \
+    | sort | uniq | xargs -L 1 -I {} ./image-pull-push.sh {} $(REGISTRY)
+
+pull-charts:
+	curl -L https://github.com/weaveworks/weave-gitops-profile-examples/releases/download/$(CERT_MANAGER)/$(CERT_MANAGER_CHART) --output $(CERT_MANAGER_CHART)
+	curl -L https://s3.us-east-1.amazonaws.com/weaveworks-wkp/releases/charts-v3/$(WGE_CHART) --output  $(WGE_CHART)
+
+push-charts:
+	helm cm-push -f $(CERT_MANAGER_CHART) $(PRIVATE_HELM_REPO_NAME)
+	helm cm-push -f $(WGE_CHART) $(PRIVATE_HELM_REPO_NAME)
+```
+where `image-pull-push.sh` is similar to
```

Comment:
Another option is [skopeo](https://www.redhat.com/en/blog/skopeo-copy-rescue) which doesn't require Docker to be running locally which might be an issue for enterprises who don't allow users to install things?

Reply:
I'm not sure we can rely on folks being able to install Kind either :-)

Reply:
agree, the doc itself is more to provide guidance than a recipe 

i tried to capture [it here](https://github.com/weaveworks/weave-gitops/pull/3006/files#diff-c807d0873ce72f2d1cd51982bddcd77f162faed3a34f2ada80c03b8b1e9968dfR49) but let me expand on that

Reply:
One reason not to use docker pull/push is that it loses the manifest.

Most of the tools that can copy images will retain the manifests, which means things like different architectures are retained, e.g. regctl https://github.com/regclient/regclient

To be clear, I have no particular attachment to skopeo (I use regctl personally) but it's at least supported by Red Hat :-)

Reply:
Expanded here https://github.com/weaveworks/weave-gitops/pull/3006/commits/fc2d3297f29d6d233f1c609a9a51edf258cabea3

Reply:
updated